raiden/transfer/node.py

```diff
@@ -745,6 +745,31 @@ def handle_receive_unlock(
     return subdispatch_to_paymenttask(chain_state, state_change, secrethash)
 
 
+def handle_contract_receive_channel_batch_unlock(
+        chain_state: ChainState,
+        state_change: ContractReceiveChannelBatchUnlock,
+) -> TransitionResult:
+    is_related = (
+        chain_state.our_address == state_change.participant or
+        chain_state.our_address == state_change.partner
+    )
+    if not is_related:
+        return TransitionResult(chain_state, [])
+
+    # First perform the token network actions so that if we are the participant
+    # the channel is deleted from channelidentifiers_to_channels
+    token_network_result = handle_token_network_action(
+        chain_state=chain_state,
+        state_change=state_change,
+    )
+    transfers_result = subdispatch_to_all_lockedtransfers(
```

Comment:
We subdispatch it to the mediator task here, but no specific processing is done in `mediator.state_transition`. We just rely on the order here so that `mediator.state_transition()` reaches the end and [enters](https://github.com/raiden-network/raiden/blob/23adf47bfbdadc6854cdb983af011909165cc94e/raiden/transfer/mediated_transfer/mediator.py#L1378) `clear_if_finalized()` to cleanup the mediator task.

Reply:
Can't we just rely on the block state change? this way we don't have to have extra code and it should clear it too

Reply:
Hmmm we couldn't before. Because the assertions would hit or we would get the key errors -- the manifestation of the bug. But right now after https://github.com/raiden-network/raiden/pull/3015/commits/66c7005bc0184125444147250dfad37b9493e783 I think it should work and is probably cleaner.